Missing prepare/unprepare

Looked at the documentation at https://github.com/sidorares/node-mysql2/blob/master/documentation/Prepared-Statements.md when I try to run a prepared statement

```js
connection.execute('statement', (error, results) => {

});

connection.unprepare('statement');
```


I get the following error

`
 error TS2339: Property 'unprepare' does not exist on type 'Connection'.
`

The prepare method also doesn't exist

--- a/index.d.ts
+++ b/index.d.ts
@@ -186,6 +186,14 @@ export interface ConnectionOptions extends mysql.ConnectionOptions {
   };
 }
 
+export interface ConnectionConfig extends ConnectionOptions {
+  mergeFlags(defaultFlags: string[], userFlags: string[] | string): number;
+  getDefaultFlags(options?: ConnectionOptions): string[];
+  getCharsetNumber(charset: string): number;
+  getSSLProfile(name: string): { ca: string[] };
+  parseUrl(url: string): { host: string, port: number, database: string, user: string, password: string, [key: string]: any };
+}
+
 export interface PoolOptions extends mysql.PoolOptions, ConnectionOptions {}
 
 export function createConnection(connectionUri: string): Connection;

--- a/index.d.ts
+++ b/index.d.ts
@@ -72,6 +72,8 @@ export interface Connection extends mysql.Connection {
   ): mysql.Query;
   ping(callback?: (err: mysql.QueryError | null) => any): void;
   promise(promiseImpl?: PromiseConstructor): PromiseConnection;
+  unprepare(sql: string): mysql.PrepareStatementInfo;
+  prepare(sql: string, callback?: (err: mysql.QueryError | null, statement: mysql.PrepareStatementInfo) => any): mysql.Prepare;
 }
 
 export interface PoolConnection extends mysql.PoolConnection, Connection {
@@ -149,6 +151,8 @@ export interface Pool extends mysql.Connection {
   on(event: 'release', listener: (connection: PoolConnection) => any): this;
   on(event: 'enqueue', listener: () => any): this;
   promise(promiseImpl?: PromiseConstructor): PromisePool;
+  unprepare(sql: string): mysql.PrepareStatementInfo;
+  prepare(sql: string, callback?: (err: mysql.QueryError | null, statement: mysql.PrepareStatementInfo) => any): mysql.Prepare;
 }
 
 type authPlugins = (pluginMetadata: {
@@ -180,14 +184,6 @@ export interface ConnectionOptions extends mysql.ConnectionOptions {
   authPlugins?: {
     [key: string]: authPlugins;
   };
-}
-
-export interface ConnectionConfig extends ConnectionOptions {
-  static mergeFlags(defaultFlags: string[], userFlags: string[] | string): number;
-  static getDefaultFlags(options?: ConnectionOptions): string[];
-  static getCharsetNumber(charset: string): number;
-  static getSSLProfile(name: string): { ca: string[] };
-  static parseUrl(url: string): { host: string, port: number, database: string, user: string, password: string, [key: string]: any };
 }
 
 export interface PoolOptions extends mysql.PoolOptions, ConnectionOptions {}

--- a/typings/mysql/index.d.ts
+++ b/typings/mysql/index.d.ts
@@ -7,7 +7,9 @@ import {PoolOptions} from './lib/Pool';
 import BasePoolCluster = require('./lib/PoolCluster');
 import {PoolClusterOptions} from './lib/PoolCluster';
 import BaseQuery = require('./lib/protocol/sequences/Query');
+import BasePrepare = require('./lib/protocol/sequences/Prepare');
 import {QueryOptions, StreamOptions, QueryError} from './lib/protocol/sequences/Query';
+import {PrepareStatementInfo} from 'mysql2/typings/mysql/lib/protocol/sequences/Prepare';
 
 export function createConnection(connectionUri: string): Connection;
 export function createConnection(config: BaseConnection.ConnectionOptions): Connection;
@@ -28,7 +30,8 @@ export {
     PoolOptions,
     PoolClusterOptions,
     QueryOptions,
-    QueryError
+    QueryError,
+    PrepareStatementInfo
 };
 export * from './lib/protocol/packets/index';
 
@@ -38,3 +41,4 @@ export interface PoolConnection extends BasePoolConnection {}
 export interface Pool extends BasePool {}
 export interface PoolCluster extends BasePoolCluster {}
 export interface Query extends BaseQuery {}
+export interface Prepare extends BasePrepare {}

--- a/typings/mysql/lib/Connection.d.ts
+++ b/typings/mysql/lib/Connection.d.ts
@@ -4,6 +4,7 @@
 // Modifications copyright (c) 2021, Oracle and/or its affiliates.
 
 import Query = require('./protocol/sequences/Query');
+import Prepare = require('./protocol/sequences/Prepare');
 import {OkPacket, FieldPacket, RowDataPacket, ResultSetHeader} from './protocol/packets/index';
 import {EventEmitter} from 'events';
 

--- a/typings/mysql/lib/protocol/sequences/Prepare.d.ts
+++ b/typings/mysql/lib/protocol/sequences/Prepare.d.ts
@@ -1,0 +1,52 @@
+import Sequence = require('mysql2/typings/mysql/lib/protocol/sequences/Sequence');
+import Query = require('mysql2/typings/mysql/lib/protocol/sequences/Query');
+import {OkPacket, FieldPacket, RowDataPacket, ResultSetHeader} from 'mysql2/typings/mysql/lib/protocol/packets';
+import {Readable} from 'stream';
+
+declare namespace Prepare {
+	export class PrepareStatementInfo {
+		close(): void;
+		execute<T extends RowDataPacket[][] | RowDataPacket[] | OkPacket | OkPacket[] | ResultSetHeader>(
+			paramaters: any | any[] | { [param: string]: any },
+			callback?: (
+				err: Query.QueryError | null,
+				result: T,
+				fields: FieldPacket[]
+			) => any): Query;
+	}
+}
+
+declare class Prepare extends Sequence {
+	/**
+	 * The SQL for a constructed query
+	 */
+	sql: string;
+
+	/**
+	 * Emits a query packet to start the query
+	 */
+	start(): void;
+
+	/**
+	 * Determines the packet class to use given the first byte of the packet.
+	 *
+	 * @param firstByte The first byte of the packet
+	 * @param parser The packet parser
+	 */
+	determinePacket(firstByte: number, parser: any): any;
+
+	/**
+	 * Creates a Readable stream with the given options
+	 *
+	 * @param options The options for the stream.
+	 */
+	stream(options?: Query.StreamOptions): Readable;
+
+	on(event: string, listener: Function): this;
+	on(event: 'error', listener: (err: Query.QueryError) => any): this;
+	on(event: 'fields', listener: (fields: FieldPacket, index: number) => any): this;
+	on(event: 'result', listener: (result: RowDataPacket | OkPacket, index: number) => any): this;
+	on(event: 'end', listener: () => any): this;
+}
+
+export = Prepare;

